File: Server/models/Clothing.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const clothingSchema = new mongoose.Schema({
  clothing_id: {
    type: Number,
    unique: true
  },
  user_id: {
    type: Number,
    required: true, 
    ref: 'User',
    index: true 
  },
  image_id: {
    type: Number, 
    required: true,
    ref: 'Image', 
    index: true
  },
  category_id: {
    type: Number,
    required: true,
    ref: 'Category', 
    index: true  
  },
  color_hex: {
    type: String,
    default: null
  },
  color_family: {
    type: String,
    default: null
  },
  material: {
    type: String,
    default: null
  },
  size: {
    type: String,
    default: null
  },
  brand_name: {
    type: String,
    default: null
  },
  is_favorite: {
    type: Boolean,
    default: false 
  },
  status: {
    type: String,
    enum: ['active', 'archived', 'in_wash'], 
    default: 'active', 
    index: true 
  },
  last_worn: {
    type: Date,
    default: null 
  }
}, {
  timestamps: true 
});

clothingSchema.pre('save', async function(next) {
  const doc = this;

  if (!doc.isNew) {
    return next();
  }

  try {
    const counter = await Counter.findByIdAndUpdate(
      { _id: 'clothing_id' }, 
      { $inc: { seq: 1 } },
      { new: true, upsert: true }
    );

    doc.clothing_id = counter.seq;
    next();
  } catch (error) {
    next(error);
  }
});

module.exports = mongoose.model('Clothing', clothingSchema, 'clothes');

File: Android/app/src/main/java/com/example/smartfashion/model/Clothing.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Clothing(
    @SerializedName("clothing_id")
    val clothingId: Int? = null,

    @SerializedName("user_id")
    val userId: Int,

    @SerializedName("image_id")
    val imageId: Int,

    @SerializedName("category_id")
    val categoryId: Int,

    @SerializedName("color_hex")
    val colorHex: String? = null,

    @SerializedName("color_family")
    val colorFamily: String? = null,

    @SerializedName("material")
    val material: String? = null,

    @SerializedName("size")
    val size: String? = null,

    @SerializedName("brand_name")
    val brandName: String? = null,

    @SerializedName("is_favorite")
    val isFavorite: Boolean = false,

    @SerializedName("status")
    val status: String = "active",

    @SerializedName("last_worn")
    val lastWorn: Date? = null,

    @SerializedName("createdAt")
    val createdAt: Date? = null,

    @SerializedName("updatedAt")
    val updatedAt: Date? = null
)